### DISCUSSION TITLE
Get modification timestamp for CSV/JSON files

### DISCUSSION THREAD

**Discussion by indigoviolet** [duckdb/duckdb#9902] [Q&A]
When I read a glob of files using read_csv_auto etc, can I get the modification timestamp or other metadata for those files as a column somehow?

**Comment by pdet** [comment]
Yes.
You can use the `sniff_csv` function. It should return the timestamp format. See the details [here](https://duckdb.org/docs/data/csv/auto_detection.html).

Obs: This function is only available in the dev version.

**Reply by indigoviolet** [reply]
This seems to be sniffing the format for timestamp columns within a CSV file; I'm referring to filesystem metadata about the CSV file - mtime, ctime etc.

**Reply by pdet** [reply]
Ah, I got you now. I don't think we have functions exposing fs metadata available.

**Reply by indigoviolet** [reply]
Ok, thanks! Please consider this a feature request :)

**Reply by ned2** [reply]
+1 to this feature request. I've bumped into this need when consuming a pile of JSON line files from an auto generated source that contain duplicate records, and need to resolve by selecting the record from the most recently created JSON file.
